cloud/terraform/azure_config_builder_v2.py

```diff
@@ -0,0 +1,332 @@
+
+        return self.resources_tf
+
+    def __parse_vhd_name(self, vhd_uri):
+        """Parse a vhd name and return extracted image details
+
+        Regex101: https://regex101.com/r/almD3W/1
+
+        Args:
+            vhd_uri: Image uri.
+        Returns:
+            Dictionary with additional information about the vhd.
+        """
+        #     product_name = one of the EXD-SP defined Azure product names e.g. rhel-sap-azure
+        #     See https://docs.google.com/document/d/19EiQ-XyvsUo4r-u-SFetVJ_P8M96mwtEyuuDKpqMb0c/edit#
+        #     for Stratosphere naming conventions.
+        #     version = RHEL version in MAJOR.MINOR(.PATCH)
+        #     date = date image was produced
+        #     build_nr = integer of builditeration performed on the build date
+        #     arch = architecture
+        vhd_name = vhd_uri.split('/')[-1]
+        azure_vhd_regex = (
+            r"(?P<product_name>\D*)\-(?P<version>[\d]+\.[\d]+(?:\.[\d]+)?)\-"
+            r"(?P<date>\d{4}\d{2}\d{2})\.\bsp\.(?P<build_nr>\d)\.(?P<arch>\S*)\.\bvhd"
+        )
+        matches = re.match(azure_vhd_regex, vhd_name, re.IGNORECASE)
+        if matches:
+            return matches.groupdict()
+
+        return {}
+
+    def __new_azure_shared_image(self, instance, vhd_properties):
+        product = vhd_properties['product_name']
+        version = vhd_properties['version'].replace(".", "-")
+
+        if vhd_properties['arch'] == 'x86_64':
+            arch = 'x64'
+        elif vhd_properties['arch'] == 'aarch64':
+            arch = 'Arm64'
+        else:
+            arch = 'x64'
+
+        name = self.create_resource_name([f"{product}-{version}-{arch}"])
+        instance['azurerm_shared_image'] = name
+
+        identifier = {
+            'publisher': self.default_publisher,
+            'offer': product.upper(),
+            'sku': version
+        }
+
+        new_image_definition = {
+            'name': name,
+            'gallery_name': self.gallery_name,
+            'resource_group_name': self.resource_group,
+            'location': instance['location'],
+            'os_type': 'Linux',
+            'identifier': identifier,
+            'hyper_v_generation': 'V2',
+            'architecture': arch
+        }
+
+        self.resources_tf['resource']['azurerm_shared_image'][name] = new_image_definition
+
+    def __new_azure_shared_image_version(self, instance, vhd_properties):
+        name = self.create_resource_name([instance['azurerm_shared_image'], 'img-version'])
+        instance['azurerm_shared_image_version'] = name
+
+        target_region = {
+            'name': 'eastus',
+            'regional_replica_count': 1
+        }
+
+        new_image = {
+            'name': '0.0.1',
+            'blob_uri': instance['vhd_uri'],
+            'storage_account_id': f"/subscriptions/{self.subscription_id}/resourceGroups/rh-resource/providers/Microsoft.Storage/storageAccounts/rhimages",
```

Comment:
Is there a way that we can compose this URI without having to hardcode the storage account and other things?

Reply:
Hm. Technically we could get all the information we need to build any image from the VHD uri, so yes we could extract it as part of the image name parsing as well. Otherwise we would have to set that as part of the parameters for the pipeline request. What do you think?